pytests/tests/test_cfgreader.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2019-2021 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the GNU General Public License v2 (the "License");
+# you may not use this file except in compliance with the License. The terms
+# of the License are located in the COPYING file of this distribution.
+#
+
+import os
+import pytest
+
+NEW_REPO_FN = 'test.repo'
+ORIG_REPO_FN = 'photon-test.repo'
+
+
+def create_repoconf(fn):
+    content = '''
+[cfg-test]
+name=    Test Repo
+baseurl = http://localhost:8080/photon-test
+enabled      =1
+gpgcheck=0
+'''
+    with open(fn, 'w') as f:
+        f.write(content)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_test(utils):
+    global NEW_REPO_FN
+    global ORIG_REPO_FN
+
+    NEW_REPO_FN = os.path.join(utils.config['repo_path'], 'yum.repos.d',
+                               NEW_REPO_FN)
+
+    ORIG_REPO_FN = os.path.join(utils.config['repo_path'], 'yum.repos.d',
+                                ORIG_REPO_FN)
+
+    create_repoconf(NEW_REPO_FN)
+
+    # take backup of original repo file
+    os.rename(ORIG_REPO_FN, ORIG_REPO_FN+'.bak')
```

Comment:
Why not just use the `-c`/`--config` option instead of moving the config away?

Reply:
This is not config, this is repo file. If I need to use a different config, I need to create both config and repo file. This is easier.

Reply:
Oh sorry, I was confused. But then - we could do the test with the config file too.

Reply:
We can do it but it will be redundant right? The goal here is giving some tricky but valid inputs to config reader. This is doing exactly that.